### ISSUE TITLE
Collection link not available for non-listable resources (/publish)

### ISSUE THREAD

**Issue by cjellick** [rancher/cattle#254]
In this schema:
```
{
  "id": "publish",
  "type": "schema",
  "links": {
    "self": "…/v1/schemas/publish",
  },
  ...
  "collectionMethods": [
    "POST",
  ],
}
```
The POST collection method is available, but no collection link is provided.

**Comment by cjellick** [comment]
Also, POSTing to /publish results in an empty response. For consistency, houldn't it return the publish resource that was posted?

**Comment by deniseschannon** [comment]
Closed in favor of rancher/rancher#3857
